joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForFunctionsCreator.scala

```diff
@@ -0,0 +1,49 @@
+package io.joern.gosrc2cpg.astcreation
+
+import io.joern.gosrc2cpg.parser.ParserAst.BlockStmt
+import io.joern.gosrc2cpg.parser.{ParserKeys, ParserNodeInfo}
+import io.joern.x2cpg.Ast
+import io.joern.x2cpg.utils.NodeBuilders.newMethodReturnNode
+import io.shiftleft.codepropertygraph.generated.nodes.NewMethodParameterIn
+import ujson.Value
+
+trait AstForFunctionsCreator { this: AstCreator =>
+  def astForFuncDecl(funcDecl: ParserNodeInfo) = {
+
+    val filename       = "fileName"
```

Comment:
This looks incomplete for now. I would also suggest not putting any placeholders into fields that aren't filled out but rather leaving them. This will otherwise confuse later stages, e.g., the `CallLinker`.

Reply:
Point noted, will take this into consideration in upcoming PRs